python/STS_ENABLED_IN_REGIONS/STS_ENABLE_IN_REGIONS.py

```diff
@@ -0,0 +1,425 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+'''
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  STS_ENABLE_IN_REGIONS
+Description:
+  Verify that STS endpoints are enabled in whitelisted regions.
+Trigger:
+  Configuration change on AWS::IAM::Role
+Resource Type to report on:
+  AWS::IAM::Role
+Rule Parameters:
+  ActivateRegions
+    Verify Security Token Service is activated in whitelisted region(s).
+Scenarios:
+  Scenario 1:
+  Given: The same number of active and specified regions
+    Then: Return COMPLIANT
+  Scenario 2:
+  Given: There are more active regions then specified.
+    Then: Return NON_COMPLIANT
+  Scenario 3:
+  Given: There are more active regions then specified.
+    Then: Return NON_COMPLIANT
+'''
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::IAM::Role'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    account_id = valid_rule_parameters["account_id"]
+    role = valid_rule_parameters["role"]
+    session = boto3.Session()

```

Comment:
Bit worried about this, in the case of a cross account assume role using the ConfigRole role. TThis rule is going to work only in the same account right now, if I understand well.

Reply:
The current assumed role is not the config’s role but a new role which is the role of the Lambda. The role is working on same account as the config is run. Would need a bit more clarification

Reply:
If we move this rule into the console, we need the Config Role to be able to do the job.